src/Chapter3.hs

```diff
@@ -560,22 +650,38 @@ introducing extra newtypes.
 🕯 HINT: if you complete this task properly, you don't need to change the
     implementation of the "hitPlayer" function at all!
 -}
+
+newtype Damage = Damage
+    {
+      dmg :: Int
+    }
+
+newtype Defense = Defense
+    {
+      def :: Int
+    }
+
+newtype HitHealth = HitHealth
+    {
+      hh :: Int
+    }
+
 data Player = Player
-    { playerHealth    :: Int
+    { playerHealth    :: HitHealth
     , playerArmor     :: Int
     , playerAttack    :: Int
     , playerDexterity :: Int
     , playerStrength  :: Int
     }
 
-calculatePlayerDamage :: Int -> Int -> Int
-calculatePlayerDamage attack strength = attack + strength
+calculatePlayerDamage :: Int -> Int -> Damage
+calculatePlayerDamage attack strength = Damage (attack + strength)
 
-calculatePlayerDefense :: Int -> Int -> Int
-calculatePlayerDefense armor dexterity = armor * dexterity
+calculatePlayerDefense :: Int -> Int -> Defense
+calculatePlayerDefense armor dexterity = Defense (armor * dexterity)
 
-calculatePlayerHit :: Int -> Int -> Int -> Int
-calculatePlayerHit damage defense health = health + defense - damage
+calculatePlayerHit :: Damage -> Defense -> HitHealth -> HitHealth
+calculatePlayerHit damage defense hithealth = HitHealth (hh hithealth + def defense - dmg damage)
```

Comment:
Nice work on newtypes! :medal_sports: Newtypes are one very important feature in Haskell :+1:
You can read more about the newtype pattern in our blog post: https://kowainik.github.io/posts/haskell-mini-patterns#newtype